salt/metalk8s/orchestrate/deploy_node.sls

```diff
@@ -113,6 +115,42 @@ Wait minion available:
 
 {%- endif %}
 
+{%- if 'etcd' in roles and 'etcd' not in skip_roles %}
+
+Prepare etcd dependencies:
+  salt.state:
+    - tgt: {{ node_name }}
+    - saltenv: metalk8s-{{ version }}
+    - sls:
+      - metalk8s.roles.internal.node-without-calico
+      - metalk8s.kubernetes.etcd.prepared
+    - require_in:
+      - salt: Run the highstate
+
+Register the node into etcd cluster:
+  salt.runner:
+    - name: state.orchestrate
+    - pillar: {{ pillar | json  }}
+    - mods:
+      - metalk8s.orchestrate.register_etcd
+    - require:
+      - salt: Prepare etcd dependencies
+    - require_in:
+      - salt: Run the highstate
+      - salt: Install and wait for healthy etcd member
```

Comment:
ditto, and you don't need to `require_in` on `Install and wait for healthy etcd member` since you already `require` it in next state